cumulus_library/apis/umls.py

```diff
@@ -91,16 +95,32 @@ def get_vsac_valuesets(
             url = url.replace("http:", "https:")
         if "/res/" not in url:
             url = url.replace("/fhir/", "/fhir/res/")
-        response = self.session.get(url)
+        match action:
+            case "definition":
+                response = self.session.get(url)
+            case "expansion":
+                response = self.session.get(url + f"?offset={offset}")
         if response.status_code == 404:
             raise errors.ApiError(f"Url not found: {url}")
         all_responses = [response.json()]
+        if action == "expansion":
+            # Do we need to fetch the next page?
+            # Note: the VSAC API hard codes the page size to 1000
+            # and ignores the count param in the FHIR API
+            # (https://www.hl7.org/fhir/ValueSet-operation-expand.html),
+            # so we just hardcode an expansion of 1000 and hope it
+            # doesn't chage
+            if all_responses[0]["expansion"]["total"] > offset + 1000:
```

Comment:
That `total` value is consistent as you walk through the results? i.e. the second response's total value isn't like "total left to go" or anything?

Reply:
correct, it's always the dataset full size.